iphone/Maps/UI/Editor/MWMEditorViewController.mm

```diff
@@ -724,6 +724,18 @@ - (void)fillCell:(UITableViewCell * _Nonnull)cell atIndexPath:(NSIndexPath * _No
   }
 }
 
+- (void)removeAdditionalNameAtIndexPath:(NSIndexPath *)indexPath
+{
+  MWMEditorAdditionalNameTableViewCell * cell = [self.tableView cellForRowAtIndexPath:indexPath];
+  if (!cell)
+    return;
+  [self cell:cell changedText:@""];
+  m_mapObject.RemoveNeedlessNames();
+  m_newAdditionalLanguages.erase(std::remove(m_newAdditionalLanguages.begin(), m_newAdditionalLanguages.end(), cell.code), m_newAdditionalLanguages.end());
+  [self configTable];
```

Comment:
Why the config is needed? Maybe some comments would be helpful.

Reply:
After the `RemoveNeedlessNames` the `[self configTable];` will reload the available for the adding languages list (this is very complex reloading).
There are 2 important steps:
1. remove the language from the `m_newAdditionalLanguages` so it will not be visible on the Editor screen
2. _return_ this language to the available list so it will be visible in the Add new lang screen

Reloading will not affect other edited fields and will not erase any data.